### ISSUE TITLE
Styles as supported by GITLAB and Shields.io

### ISSUE THREAD

**Issue by krishvk** [psincraian/pepy#208]
Is it possible to use/support styles like other alternatives (GItlab and Shields.io)

![image](https://user-images.githubusercontent.com/7437418/80856175-3b23aa00-8c65-11ea-8494-14f6591a4c93.png)

**Comment by KrishKasula** [comment]
![Uploading image.png…]()

**Comment by psincraian** [comment]
Hi,

Thanks for the suggestion!

It's possible but it will require some work, I will try to prioritize it based on the demand for it 😊

**Comment by hugovk** [comment]
Would it be possible for shields.io to provide the badges? They have many different options.

**Comment by ffe4** [comment]
@hugovk there is a guide on the shields GitHub repo on [how to add a badge for a service](https://github.com/badges/shields/blob/master/doc/TUTORIAL.md)

**Comment by finnmglas** [comment]
I suppose that it would be useful to at least implement the common GET arguments that https://shields.io implements:

`label=<LABEL>&message=<MESSAGE>&color=<COLOR>` plus the style argument

**Comment by finnmglas** [comment]
@krishvk Here is a inofficial implementation based on a comment by @psincraian : 

plastic:

![](https://img.shields.io/badge/dynamic/json?style=plastic&color=97ca00&maxAge=86400&label=downloads&query=%24.total_downloads&url=https%3A%2F%2Fapi.pepy.tech%2Fapi%2Fprojects%2Frequests)

flat:

![](https://img.shields.io/badge/dynamic/json?style=flat&color=97ca00&maxAge=86400&label=downloads&query=%24.total_downloads&url=https%3A%2F%2Fapi.pepy.tech%2Fapi%2Fprojects%2Frequests)

flat-square:

![](https://img.shields.io/badge/dynamic/json?style=flat-square&color=97ca00&maxAge=86400&label=downloads&query=%24.total_downloads&url=https%3A%2F%2Fapi.pepy.tech%2Fapi%2Fprojects%2Frequests)

social:

![](https://img.shields.io/badge/dynamic/json?style=social&color=97ca00&maxAge=86400&label=downloads&query=%24.total_downloads&url=https%3A%2F%2Fapi.pepy.tech%2Fapi%2Fprojects%2Frequests)

generally something like this (don't forget to adjust the parameters, style etc):

```
![](https://img.shields.io/badge/dynamic/json?style=<STYLE>&label=downloads&query=%24.total_downloads&url=https%3A%2F%2Fapi.pepy.tech%2Fapi%2Fprojects%2Frequests)
```

---

I am not sure if this helps you, but it helped me ^^

**Comment by WenjieDu** [comment]
> @krishvk Here is a inofficial implementation based on a comment by @psincraian :
> 
> plastic:
> 
> ![](https://camo.githubusercontent.com/c3892d832884b79e147445cb19d899b94544bebd98836d25df8ef52609d542e2/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f64796e616d69632f6a736f6e3f7374796c653d706c617374696326636f6c6f723d393763613030266d61784167653d3836343030266c6162656c3d646f776e6c6f6164732671756572793d2532342e746f74616c5f646f776e6c6f6164732675726c3d68747470732533412532462532466170692e706570792e7465636825324661706925324670726f6a656374732532467265717565737473)
> 
> flat:
> 
> ![](https://camo.githubusercontent.com/f9f91549aafe0bfc304aca7c0897529dbf97e2f8696d02061a9184678c9ac74e/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f64796e616d69632f6a736f6e3f7374796c653d666c617426636f6c6f723d393763613030266d61784167653d3836343030266c6162656c3d646f776e6c6f6164732671756572793d2532342e746f74616c5f646f776e6c6f6164732675726c3d68747470732533412532462532466170692e706570792e7465636825324661706925324670726f6a656374732532467265717565737473)
> 
> flat-square:
> 
> ![](https://camo.githubusercontent.com/444dc13d4dc1b75bbab97e6d10b8f3d5feddeb39846af5cb6be02c35f0a48378/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f64796e616d69632f6a736f6e3f7374796c653d666c61742d73717561726526636f6c6f723d393763613030266d61784167653d3836343030266c6162656c3d646f776e6c6f6164732671756572793d2532342e746f74616c5f646f776e6c6f6164732675726c3d68747470732533412532462532466170692e706570792e7465636825324661706925324670726f6a656374732532467265717565737473)
> 
> social:
> 
> ![](https://camo.githubusercontent.com/a8a5a975cc201b95c1b4c9c76ecaf671cb951b85ab2db0b1a75c77ac527f3c92/68747470733a2f2f696d672e736869656c64732e696f2f62616467652f64796e616d69632f6a736f6e3f7374796c653d736f6369616c26636f6c6f723d393763613030266d61784167653d3836343030266c6162656c3d646f776e6c6f6164732671756572793d2532342e746f74616c5f646f776e6c6f6164732675726c3d68747470732533412532462532466170692e706570792e7465636825324661706925324670726f6a656374732532467265717565737473)
> 
> generally something like this (don't forget to adjust the parameters, style etc):
> 
> ```
> ![](https://img.shields.io/badge/dynamic/json?style=<STYLE>&label=downloads&query=%24.total_downloads&url=https%3A%2F%2Fapi.pepy.tech%2Fapi%2Fprojects%2Frequests)
> ```
> 
> I am not sure if this helps you, but it helped me ^^

It seems that the examples provided by @finnmglas are down, showing "resource not found".

**Comment by WenjieDu** [comment]
Awesome! Thank you for your information @aMahanna 😊

**Comment by WenjieDu** [comment]
@aMahanna BTW, do you know how to make the displayed number of downloads be an abbreviation (e.g. 17K), rather than an exact number (e.g. 17342)? I tried to look for PePy documentation but failed. 🥲

**Comment by aMahanna** [comment]
@WenjieDu  So the `static.pepy.tech` API allows you to abbreviate the number with the `units=abbreviation` query parameter

[![Downloads](https://static.pepy.tech/personalized-badge/requests?period=week&units=abbreviation&left_color=black&right_color=blue&left_text=Downloads)](https://pepy.tech/project/requests)

```
[![Downloads](https://static.pepy.tech/personalized-badge/requests?period=week&units=international_system&left_color=black&right_color=blue&left_text=Downloads)](https://pepy.tech/project/requests)
```  

But unfortunately I am not sure how this query parameter would work if we are relying on the `img.shields.io` to add a custom style

**Comment by WenjieDu** [comment]
Right, that's true. `api.pepy.tech` only provides the exact number. It seems that `shields.io` doesn't offer such a function for abbreviation.

**Comment by WenjieDu** [comment]
> @WenjieDu So the `static.pepy.tech` API allows you to abbreviate the number with the `units=abbreviation` query parameter
> 
> [![Downloads](https://camo.githubusercontent.com/421729fee78663c3e5c1495126c9151757d38088255fe7d2caa2e6d3b35f6861/68747470733a2f2f7374617469632e706570792e746563682f706572736f6e616c697a65642d62616467652f72657175657374733f706572696f643d7765656b26756e6974733d616262726576696174696f6e266c6566745f636f6c6f723d626c61636b2672696768745f636f6c6f723d626c7565266c6566745f746578743d446f776e6c6f616473)](https://pepy.tech/project/requests)
> 
> ```
> [![Downloads](https://static.pepy.tech/personalized-badge/requests?period=week&units=abbreviation&left_color=black&right_color=blue&left_text=Downloads)](https://pepy.tech/project/requests)
> ```
> 
> But unfortunately I am not sure how this query parameter would work if we are relying on the `img.shields.io` API to add a custom style

Right, that's true. `api.pepy.tech` only provides the exact number. It seems that `shields.io` doesn't offer such a function for abbreviation.
